src/java.base/share/classes/jdk/internal/classfile/impl/SplitConstantPool.java

```diff
@@ -100,7 +99,6 @@ public PoolEntry entryByIndex(int index) {
 
     @Override
     public <T extends PoolEntry> T entryByIndex(int index, Class<T> cls) {
-        Objects.requireNonNull(cls);
```

Comment:
I think this was added because we want the NPE to be thrown before IAE if index is 0 but cls is null. It might be fine one way or another.